xmake/core/base/os.lua

```diff
@@ -1002,8 +1002,9 @@ end
 -- get the system null device
 function os.nuldev(input)
 
+    -- mingw and cygwin have /dev/null
     if input then
-        if os.host() == "windows" then
+        if os.host() == "windows" and not os.is_subhost("msys", "cygwin") then
```

Comment:
I know msys2 has /dev/null. But it seems to have had some problems before. The compatibility wasn't very good, so I disabled it as well.

Reply:
Also, xmake's os.exec is always a direct call to CreateProcess, which doesn't support /dev/null for msys2. xmake doesn't use mingw's fork to create processes.

Reply:
I'll remove this too.